app/scripts/models/auth_brokers/base.js

```diff
@@ -89,17 +94,44 @@ define(function (require, exports, module) {
     },
 
     /**
-     * initialize the broker with any necessary data.
+     * Initialize the broker with any necessary data.
+     *
      * @returns {Promise}
      */
     fetch () {
       return p().then(() => {
         this._isForceAuth = this._isForceAuthUrl();
         this.importSearchParamsUsingSchema(QUERY_PARAMETER_SCHEMA, AuthErrors);
+
+        if (this.hasCapability('fxaStatus')) {
+          return this._fetchFxaStatus();
+        }
       });
     },
 
     /**
+     * Request FXA_STATUS info from the UA.
+     *
+     * @returns {Promise} resolves when complete.
+     */
+    _fetchFxaStatus () {
+      const TEST_REQUEST_DELAY_MS = this.isAutomatedBrowser() ? 200 : 0;
+
+      const channel = this._notificationChannel;
+      return p().delay(TEST_REQUEST_DELAY_MS)
+        // OAuth reliers will have `service` set to the client_id, Sync to `sync`
```

Comment:
Not sure if it's just me but part of this comment reads back to front. Should it be "...will have the client_id set to service"?